### PR TITLE
Feature/issue 104

### PR DESCRIPTION
**Pull request by magniff**
- Migrate to project structure, proposed at issue-104
- Tune up .travis config a bit.

### CODE REVIEW

**Comment by magniff** [comment]
Ok, everything seems to be fine. Apparently pylint ignore path option is still broken, according to this issue https://github.com/PyCQA/pylint/issues/2541

File: .travis.yml [hunk 1]
```diff
@@ -4,15 +4,18 @@ python:
...
+  - pip install --upgrade pip
+  - pip install "pytest>=3.6"
+  - pip install codecov pytest-cov pytest-asyncio pylint
+  - python setup.py develop
 
 script:
-  - pytest --cov=./ -v
-  - pylint --rcfile=.pylintrc encryption host libp2p network peer stream_muxer transport tests
+  - pytest --cov=./libp2p tests/
+  - pylint --rcfile=.pylintrc libp2p/!(kademlia|protocol_muxer) tests
```
Comment (robzajac) [inline .travis.yml]: Good catch; we actually only wanted to exclude `kademlia` (it's an external implementation that we haven't used yet; we will likely move to a Git submodule soon (#97)). We can include `protocol_muxer` in linting after #106 lands.

File: .travis.yml [hunk 2]
```diff
@@ -4,15 +4,18 @@ python:
   - "3.6"
 
 install:
-  - pip install -r requirements.txt
-  - pip install codecov pytest pytest-cov
+  - pip install --upgrade pip
+  - pip install "pytest>=3.6"
+  - pip install codecov pytest-cov pytest-asyncio pylint
```
Comment (robzajac) [inline .travis.yml]: Could we have these dependencies in `setup.py` as well?  I can't remember or find docs about why we put them here instead.

Comment (mhchia) [reply .travis.yml]: IMO [the style](https://github.com/ethereum/py-evm/blob/master/setup.py#L31-L41) might be pretty suitable, or even put them in a separate `requirements_dev.txt`

File: .travis.yml (hunk 1 above)
Comment (zixuanzh) [reply .travis.yml]: just approved #106

**Comment by zixuanzh** [comment]
Should be good to go once `travis.yml` is updated.

File: .travis.yml (hunk 2 above)
Comment (magniff) [reply .travis.yml]: `Requirements_dev` is a good option, thumbs up!) I will have a chance to add it only at the late evening, though. Can we merge this code as is and add missing deps in a separate PR or something?

**Pull request merged**